allennlp/tango/dataset.py

```diff
@@ -72,3 +74,46 @@ def run(self, reader: DatasetReader, splits: Dict[str, str]) -> DatasetDict:  #
                 instance.index_fields(vocab)
 
         return DatasetDict(splits=instances_map, vocab=vocab)
+
+
+@Step.register("dataset_remix")
+class DatasetRemixStep(Step):
+    """
+    This step can remix splits in a dataset into new splits.
+    """
+
+    DETERMINISTIC = True
+    CACHEABLE = False  # This is so fast it's not worth caching.
+    VERSION = "001"
+
+    def run(  # type: ignore
+        self, input: DatasetDict, new_splits: Dict[str, str], keep_old_splits: bool = True
+    ) -> DatasetDict:
+        def get_slice(split_name: str) -> Sequence[Any]:
+            slice_match = re.match(r"(.*)\[([0123456789:]*)]", split_name)
```

Comment:
This won't work for something like `train[:50000] + dev[:10000]`. Is it supposed to?

Reply:
Yes, it will work. This function is only called on the parts after `.split("+")`.

Reply:
Ok I see, and train[:50000] in that case is interpreted as it should be?

Reply:
Yes. It supports full Python slice syntax on line 98. In condensed form, it does `slice(*match.split(":"))`.